File: programme/array-elements-slice/array-elements-slice.go
// Write a program to generate new array within given range (start, end) sliced from orignal array

package main

import (
	"fmt"
)

func main() {
	list := []string{"abcd", "efgh", "ijklmn", "opq", "rstuvw", "xyz"}
	var start int = 2
	var end int = 3

	fmt.Printf("Orignal List: ")
	fmt.Println(list)

/*
        1. "range x" will return index, value pair or iterable entity 'x'
        2. expression "variable_name[x:y]" returns a list of element within range having  index >= x and index < y
*/
	for index, value := range list {
		list [index] = value [start-1 : end]
	}
	fmt.Printf("Modified List: ")
	fmt.Println(list)
}

File: programme/array-elements-slice/array-elements-slice.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main ()
{
	//Define first vector array
	vector<string> arr1 = {"abcd", "efgh", "ijklmn", "opq", "rstuvw", "xyz"};
	cout << "Array : ";
	for (auto i : arr1)
	{
		cout << i << ' ';
	}
	cout << '\n';

	//Define range for element slice
	int start = 2, end = 3;
	cout << "Start : " << start << '\n' << "End   : " << end << '\n';
	//Create new vector array
	vector<string> arr2;

	//Iterate through every element of arr1
	for (auto i : arr1)
	{
		//Push_back sliced element to arr2 
		arr2.push_back(i.substr(start - 1, end - start + 1));
	}
	
	//Print new vector array
	cout << "Sliced Array : ";
	for (auto i : arr2)
	{
		cout << i << ' ';
	}
	cout << '\n';

	return 0;
}
